x-pack/plugins/spaces/server/lib/spaces_client.ts

```diff
@@ -196,11 +202,14 @@ export class SpacesClient {
   }
 
   private useRbac(): boolean {
-    return this.authorization && this.authorization.mode.useRbacForRequest(this.request);
+    // TODO: remove "as any" once Security is updated to NP conventions
+    return (
+      this.authorization != null && this.authorization.mode.useRbacForRequest(this.request as any)
```

Comment:
why `useRbacForRequest` uses `WeakMap` internally? `XPackInfo.feature().getLicenseCheckResults()` is a synchronous operation. Or we protect again race conditions, when calling `useRbacForRequest` can return different results?

Reply:
> we protect again race conditions, when calling useRbacForRequest can return different results?

Yep, that's the reason. We want `useRbacForRequest` to give a consistent answer for the lifetime of a request.